packages/grid/x-data-grid/src/components/base/GridOverlays.tsx

```diff
@@ -8,12 +9,20 @@ import {
 } from '../../hooks/features/rows/gridRowsSelector';
 import { useGridApiContext } from '../../hooks/utils/useGridApiContext';
 import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
-import { gridDensityTotalHeaderHeightSelector } from '../../hooks/features/density/densitySelector';
+import { getMinimalContentHeight } from '../../hooks/features/rows/gridRowsUtils';
+
+const GridOverlayWrapperRoot = styled('div')({
+  position: 'sticky', // To stay in place while scrolling
+  top: 0,
+  left: 0,
+  width: 0, // To stay above the content instead of shifting it down
```

Comment:
Should the width and height be `auto` or the size of the viewport? I'm asking because this has a `height = 0` but its child has `height > 0`?

Reply:
This specific element should have size of 0x0px and this is intentional.
If it has non-zero size, it will shift pinned columns containers, which also use `position: sticky`.